### ISSUE TITLE
triggerCharacters `[` not useful

### ISSUE THREAD

**Issue by jdtsmith** [microsoft/pyright#2734]
**Describe the bug**
The triggerCharacters are configured as "." and "[". The first makes good sense for attribute completion, but typing e.g.:

```python
x=range(10)
x[
```

causes Emacs lsp-mode completion to pop-up a huge list of unrelated classes, all possible variable names, etc.  I can see the value of attempting completion of dictionary _keys_, but completing on a bare "[" seems to only add noise. 

**To Reproduce**
Type a left bracket after a variable name. 

**Expected behavior**
Auto-triggered completion happens only when additional useful refinement is possible.

**Screenshots or Code**

Not too helpful!
![image](https://user-images.githubusercontent.com/93749/146630744-c468ea74-ba2f-4a02-8949-680df36824b0.png)

**VS Code extension or command-line**
N/A, using emacs with lsp-mode. 

Thanks.

**Comment by erictraut** [comment]
This was added to support TypedDict literal keys. It was a highly-requested feature, so I don't think we'd remove this support.

**Comment by jdtsmith** [comment]
Aha, thanks, didn't know that one.  Can you comment on how VS Code handles this trigger?   Surely the extra completion "noise" from normal dict/array/ndarray/etc. indexing wouldn't be worth the extra convenience for literal-indexed typed dicts?
